### PR TITLE
Improve server event logic

### PR DESCRIPTION
**Pull request by tjerman**
Allow the system to determine what client to use in case some are unavailable.
Currently ws and sse are supported.

### CODE REVIEW

**Comment by tjerman** [comment]
There seems to be a bit of a bug inside `reconnecting-websockets` library -- when I manually close the connection, the lib reopens it. Not a critical bug for our use case, but should be kept in mind for the future

File: src/plugins/index.js [hunk 1]
```diff
@@ -26,7 +26,7 @@ Vue.use(messaging)
 Vue.use(bus, { eventbus })
 Vue.use(system)
 Vue.use(commands, { store })
-Vue.use(ws, { eventbus })
+Vue.use(serverEvents, { eventbus })
```
Comment (darh) [inline src/plugins/index.js:29]: Let's use this opportunity and migrate away from `eventbus` (to `$root`).

File: src/plugins/serverEvents.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      },
+
+      disconnect () {
+        if (this.client) {
+          this.client.disconnect()
+        }
+      },
+
+      async determineClient (api, jwt) {
+        for (let { Client } of providers) {
```
Comment (darh) [inline src/plugins/serverEvents.js:32]: This is overdesigned. No need to support multiple client types.
As I see this, we need one client for event handling (may that be via WS for now and one for SSE when we switch) + one client for simple REST short/long polling as a fallback.

**Review by darh — COMMENTED** [review]
Let's go slowly with this -- it might be that we'll just implement REST short/long polling for now.

**Comment by darh** [comment]
SSE will have to wait. Staying on WS.

**Pull request closed without merging**